### PR TITLE
8270803: Reduce CDS API verbosity

### PR DESCRIPTION
**Pull request by iklam**
The CDS APIs have some excessively long function/field/enum names. E.g.,

```
HeapShared::set_closed_archive_heap_region_mapped()
HeapShared::fixup_mapped_heap_regions() 
MetaspaceShared::first_closed_archive_heap_region 
FileMapInfo::write_archive_heap_regions()
```

Words such as `archive_heap_`, `archive_`, `heap_`, ... can be removed when the context is clear:

```
HeapShared::set_closed_regions_mapped()
HeapShared::fixup_mapped_regions() 
MetaspaceShared::first_closed_heap_region 
FileMapInfo::write_heap_regions()
```

This RFE also fixes inconsistency and grammatical errors in the code:

- The code uses different names like `GrowableArray<MemRegion>* {archive, mem, range}` to refer to the same thing. Change these to  `GrowableArray<MemRegion>* regions`
- Change the word `region` to `regions` when appropriate
- Remove unused field `HeapShared::_archive_heap_region_fixed`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8270803](https://bugs.openjdk.java.net/browse/JDK-8270803): Reduce CDS API verbosity

### Reviewers
 * [Yumin Qi](https://openjdk.java.net/census#minqi) (@yminqi - **Reviewer**) ⚠️ Review applies to 833bb4242992d27f5ebacd7d7423689145d7f627
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to 847965518937e1b881ed4d9d77a648fb1ddef23d

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4834/head:pull/4834` \
`$ git checkout pull/4834`

Update a local copy of the PR: \
`$ git checkout pull/4834` \
`$ git pull https://git.openjdk.java.net/jdk pull/4834/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4834`

View PR using the GUI difftool: \
`$ git pr show -t 4834`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4834.diff">https://git.openjdk.java.net/jdk/pull/4834.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/cds/filemap.cpp [hunk 1]
```diff
@@ -1918,20 +1918,20 @@ void FileMapInfo::map_heap_regions_impl() {
   assert(is_aligned(relocated_closed_heap_region_bottom, HeapRegion::GrainBytes),
          "must be");
 
-  // Map the closed_archive_heap regions, GC does not write into the regions.
-  if (map_heap_data(&closed_archive_heap_ranges,
-                    MetaspaceShared::first_closed_archive_heap_region,
-                    MetaspaceShared::max_closed_archive_heap_region,
-                    &num_closed_archive_heap_ranges)) {
-    HeapShared::set_closed_archive_heap_region_mapped();
+  // Map the closed_heap regions, GC does not write into the regions.
```
Comment (yminqi) [inline src/hotspot/share/cds/filemap.cpp:1921]: either add '_' between heap and regions or replace the "_' with space.

**Review by yminqi — APPROVED** [review]
LGTM. tiny nits.

File: src/hotspot/share/cds/filemap.cpp [hunk 2]
```diff
@@ -2018,44 +2018,44 @@ bool FileMapInfo::map_heap_data(MemRegion **heap_mem, int first,
...
       return false;
     }
   }
 
   cleanup._aborted = false;
   // the shared heap data is mapped successfully
-  *heap_mem = regions;
-  *num = region_num;
+  *regions_ret = regions;
+  *num_ret = region_num;
```
Comment (calvinccheung) [inline src/hotspot/share/cds/filemap.cpp:2030]: Instead of `num_ret`, I `num_regions` is easier to understand.

**Review by calvinccheung — APPROVED** [review]
Looks good. Just one minor suggestion.

File: src/hotspot/share/cds/filemap.cpp (hunk 2 above)
Comment (iklam) [reply src/hotspot/share/cds/filemap.cpp]: I renamed it to `num_regions_ret` to be consistent with `regions_ret`: both of them are outgoing parameters. Now their names are also consistent with the globals like `closed_heap_regions` and `num_closed_heap_regions`.

I also rearranged the parameter order to put the outgoing ones at the end.

File: src/hotspot/share/cds/filemap.cpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/cds/filemap.cpp]: I changed to `// Map the closed heap regions: GC does not write into these regions.` I also updated the comment for the open heap regions a few lines below.

**Review by calvinccheung — APPROVED** [review]

**Comment by iklam** [comment]
Thanks @calvinccheung and @yminqi for the review
/integrate

**Pull request closed without merging**